src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1784,9 +1784,7 @@ private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression oper
                 ImmutableArray<MethodSymbol> originalUserDefinedConversions = conversion.OriginalUserDefinedConversions;
                 if (originalUserDefinedConversions.Length > 1)
                 {
-                    // Method groups always have null types
-                    var operandType = (operand.Kind == BoundKind.MethodGroup) ? (object)MessageID.IDS_SK_METHOD.Localize() : operand.Type;
-
+                    var operandType = operand.Type ?? operand.Display;
```

Comment:
`operand.Display` already returns the type when it is non-`null`. You should be able to just use `operand.Display`.